src/envoy/auth/http_request.cc

```diff
@@ -0,0 +1,117 @@
+/* Copyright 2017 Istio Authors. All Rights Reserved.
```

Comment:
I think adding this function (I know it is similar to mixer's transport pattern) is over kill in this filter, as we only issue at most 1 outgoing request per incoming request. Just having a class inherits AsyncClient::Callbacks inside the filter instance is way more simpler, and also less raw pointer management would be involved.

Reply:
I don't want to limit our self only making 1 remote call per request.  You never know,  we may need more than 1 remote calls in the future.    Having a separate class is cleaner.

Reply:
Having multiple inner classes (in case we need different remote calls) is still cleaner than passing around lambdas and easier to read, and less raw pointer management. Unlike mixer client, we don't need abstract the transport.

Reply:
The key benefit of passing a transport interface to Controller object is for unit-test.  I can use a mock transport to unit-test Controller.

Reply:
Just use MockAsyncClient for unit-tests.

Reply:
I only find grpc::MockAsyncClient,  not http one.   It is not easy to use either. 
I like to keep the current way.  It is a clean separation between JWT logic and environment.  transport is environment.

Reply:
Http::MockAsyncClient is here:
https://github.com/envoyproxy/envoy/blob/master/test/mocks/http/mocks.h#L335